### DISCUSSION TITLE
listen/subscribe to a particular action?

### DISCUSSION THREAD

**Discussion by marcebdev** [ai/nanostores#132] [Q&A] [answered]
I know that actions have names, and was wondering if it's possible to subscribe to a particular action, such that you can cause a side effect when it's ran.

This might sound counter-intuitive on the surface however, it's useful in areas like form validation. Typically with many form validation libs, such as `react-hook-form` for example, you cannot sync the `value` field of an input directly to global state since it's being managed by the validation lib. In such a case there's usually an escape hatch, in the case of `react-hook-form` it's `setValue` which you get from the `useForm` hook.

Since the context of the `useForm` hook is limited in scope to the form component I see only 2 ways of being able to call `setValue` in the form component

1) react to the `updateForm` action I created by subscribing or listening to it in some way
2) use an event bus as a workaround

**Comment by ai** [comment, marked as answer]
`action()` in Nano Stores is just a syntax sugar for logger.

All logic should be in stores, and you should subscribe to some specific store changes.

For form validation, I mostly used `form` store. In `useEffect` I create a new `form` store and propagate it down by context. This store has values and validation status.

**Reply by marcebdev** [reply]
While I think it could be useful to subscribe or listen filtering by an action name in the context of React & other UI libs I think this makes sense. This did solve my issue though I think in some circumstances this could become complicated.
